service/src/test/java/uk/gov/hmcts/reform/fpl/service/RepresentativeCaseRoleServiceTest.java

```diff
@@ -0,0 +1,362 @@
+                Map.of(originalRepresentative.getEmail(), emptySet()));
+        }
+
+        @Test
+        void shouldReturnCaseRoleUpdatesWhenServingPreferenceChangedFromDigitalServiceToPost() {
+            Representative originalRepresentative = Representative.builder()
+                .role(REPRESENTING_RESPONDENT_1)
+                .email("representative@test.com")
+                .servingPreferences(DIGITAL_SERVICE)
+                .build();
+
+            Representative updatedRepresentative = fromRepresentative(originalRepresentative)
+                .servingPreferences(POST)
+                .build();
+
+            assertCaseRoleUpdates(
+                List.of(updatedRepresentative),
+                List.of(originalRepresentative),
+                Map.of(originalRepresentative.getEmail(), emptySet()));
+        }
+
+        @Test
+        void shouldReturnCaseRoleUpdatesWhenServingPreferenceChangedFromEmailToDigitalService() {
+            Representative originalRepresentative = Representative.builder()
+                .role(REPRESENTING_RESPONDENT_1)
+                .email("representative@test.com")
+                .servingPreferences(EMAIL)
+                .build();
+
+            Representative updatedRepresentative = fromRepresentative(originalRepresentative)
+                .servingPreferences(DIGITAL_SERVICE)
+                .build();
+
+            assertCaseRoleUpdates(
+                List.of(updatedRepresentative),
+                List.of(originalRepresentative),
+                Map.of(originalRepresentative.getEmail(), Set.of(SOLICITOR)));
+        }
+
+        @Test
+        void shouldReturnCaseRoleUpdatesWhenServingPreferenceChangedFromPostToDigitalService() {
+            Representative originalRepresentative = Representative.builder()
+                .role(REPRESENTING_RESPONDENT_1)
+                .servingPreferences(POST)
+                .build();
+
+            Representative updatedRepresentative = fromRepresentative(originalRepresentative)
+                .email("representative@test.com")
+                .servingPreferences(DIGITAL_SERVICE)
+                .build();
+
+            assertCaseRoleUpdates(
+                List.of(updatedRepresentative),
+                List.of(originalRepresentative),
+                Map.of(updatedRepresentative.getEmail(), Set.of(SOLICITOR)));
+        }
+    }
+
+    private void assertCaseRoleUpdates(List<Representative> updatedRepresentatives,
+                                       List<Representative> originalRepresentatives,
+                                       Map<String, Set<CaseRole>> expectedUpdates) {
+        Map<String, Set<CaseRole>> actualUpdates = representativesChangeService
+            .calculateCaseRoleUpdates(updatedRepresentatives, originalRepresentatives);
+
+        assertThat(actualUpdates).isEqualTo(expectedUpdates);
+    }
+
+    private Representative.RepresentativeBuilder fromRepresentative(Representative representative) {
```

Comment:
There is annotation ToBuilder if required. But it pollutes the code.

Reply:
did not want to change production code for test convinience